sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -137,6 +137,11 @@ def _create_app(self, cls):
 
         return app
 
+    @property
+    def token_refresh_offset(self):
+        # type: (None) -> int
+        return DEFAULT_REFRESH_OFFSET
```

Comment:
By default this is up to MSAL, which may or may not share our default behavior. MSAL does allow forcing a refresh during silent authentication, so I think these credentials could allow configuring the offset. Do you intend to add that in this PR?

Reply:
As we discussed, configuration offset is supported only in ctor. This property is for policy to get the offset setting.

Reply:
Sure, but my point is that credentials built on MSAL don't necessarily observe our `DEFAULT_REFRESH_OFFSET` because when they silently authenticate, MSAL decides whether to acquire a new access token.

Reply:
Yes. And my takeaway is it can just return something to make the contract happy. BearerTokenPolicy will keep current behavior.

Reply:
But the contract is optional. Implementing it on a credential implies that credential supports the configurable offset feature. These ones do not, so why implement the contract on them?

Though as I wrote above, I think these credentials could support the feature; do you plan to add it?

Reply:
I added this for EnvironmentCredential, ClientSecretCredential & CertificateCredential have token_refresh_offset, if UsernamePasswordCredential does not have one, it will increase consumer's burden. 

So I gave it an default one.